src/common/hydrogenic.jl

```diff
@@ -0,0 +1,23 @@
+@doc raw"""
+Radial functions from solutions of Hydrogenic Schrödinger equation.
+Same as Wannier90 user guide Table 3.3.
+# Arguments
+- `r`: radial grid
+- `n`: principal quantum number
+- `α`: diffusivity, ``\frac{Z}/{a}`` where ``Z`` is the atomic number and
+    ``a`` is the Bohr radius.
+"""
+function radial_hydrogenic(r::AbstractVector{T}, n::Integer, α::Real=one(T)) where {T<:Real}
```

Comment:
Have you checked the type stability eplicitly. This is easy to miss. Just include a test with `@test_inferred` or something like (julia docs on Unit Tests) and a call based on a float32 number.

Reply:
I did. Added a test as well (the vscode integration is very nice...)